packages/cloud-functions/src/cico/moonpayWebhook.ts

```diff
@@ -0,0 +1,57 @@
+import crypto from 'crypto'
+import * as functions from 'firebase-functions'
+import { MOONPAY_WEBHOOK_KEY } from '../config'
+import { saveTxHashProvider } from '../firebase'
+import { Provider } from './Provider'
+
+const MOONPAY_SIGNATURE_HEADER = 'Moonpay-Signature-V2'
+
+function verifyMoonPaySignature(signatureHeader: string | undefined, body: string) {
+  if (!signatureHeader) {
+    return false
+  }
+
+  const [first, second] = signatureHeader.split(',')
+  const timestamp = first.split('=')[1]
+  const signature = second.split('=')[1]
+  const signedPayload = `${timestamp}.${body}`
+
+  const hmac = crypto.createHmac('sha256', MOONPAY_WEBHOOK_KEY)
+  hmac.write(signedPayload)
+  hmac.end()
```

Comment:
Does using `.end` have performance benefits? I don't think we are doing it for the Moonpay URL signing and wondering if we should

Reply:
I have no idea really, I took this code from somewhere on the internet and assumed it was necessary :P

Reply:
Looks like we are using `.digest` for the moonpay stuff and `.end` is not necessary when that's used